### PR TITLE
sort before closing idle connections

### PR DESCRIPTION
**Pull request by nitincchauhan**

### CODE REVIEW

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 1]
```diff
@@ -607,21 +609,31 @@ public void run()
...
-                  break;
-               }
-               if (clockSource.elapsedMillis(poolEntry.lastAccessed, now) > idleTimeout) {
-                  if (connectionBag.reserve(poolEntry)) {
-                     closeConnection(poolEntry, "(connection passed idleTimeout)");
-                     removable--;
+            if (removable > 0) {
+               logPoolState("Before cleanup\t");
+               //sort pool entries on lastAccessed
+               Collections.sort(notInUseList, new Comparator<PoolEntry>() {
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: Can you try making ``PoolEntry`` a ``java.lang.Comparable`` to see if it makes this cleaner?

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 2]
```diff
@@ -607,21 +609,31 @@ public void run()
...
+                  @Override
+                  public int compare(PoolEntry pe1, PoolEntry pe2) {
+                     return pe1.lastAccessed < pe2.lastAccessed
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: Or use ``Integer.compare(pe1.lastAccesses, pe2.lastAccessed)``?

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java (hunk 1 above)
Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: sure. I thought you may not like pool entry implement default sort on last access.

Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: That is probably correct, better is the suggestion below...

File: src/main/java/com/zaxxer/hikari/pool/HikariPool.java [hunk 3]
```diff
@@ -607,21 +609,31 @@ public void run()
...
-                  if (connectionBag.reserve(poolEntry)) {
-                     closeConnection(poolEntry, "(connection passed idleTimeout)");
-                     removable--;
+            if (removable > 0) {
+               logPoolState("Before cleanup\t");
+               //sort pool entries on lastAccessed
+               Collections.sort(notInUseList, new Comparator<PoolEntry>() {
+                  @Override
+                  public int compare(final PoolEntry entryOne, final PoolEntry entryTwo) {
+                     return entryOne.lastAccessed < entryTwo.lastAccessed
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: Use Integer.compare() ...

Comment (nitincchauhan) [inline src/main/java/com/zaxxer/hikari/pool/HikariPool.java]: ok ;)

**Comment by nitincchauhan** [comment]
done now. not sure why test fails and passes randomly !

**Pull request closed without merging**